contrib/linux-kernel/lib/zstd/fse_decompress.c

```diff
@@ -288,16 +290,24 @@ size_t FSE_decompress_usingDTable(void *dst, size_t originalSize, const void *cS
 	return FSE_decompress_usingDTable_generic(dst, originalSize, cSrc, cSrcSize, dt, 0);
 }
 
-size_t FSE_decompress_wksp(void *dst, size_t dstCapacity, const void *cSrc, size_t cSrcSize, FSE_DTable *workSpace, unsigned maxLog)
+size_t FSE_decompress_wksp(void *dst, size_t dstCapacity, const void *cSrc, size_t cSrcSize, unsigned maxLog, void *workspace, size_t workspaceSize)
 {
 	const BYTE *const istart = (const BYTE *)cSrc;
 	const BYTE *ip = istart;
-	short counting[FSE_MAX_SYMBOL_VALUE + 1];
+	FSE_DTable *dt = (FSE_DTable *)workspace;
+	short *counting = (short *)(dt + FSE_DTABLE_SIZE_U32(maxLog));
+	BYTE *wkspEnd = (BYTE *)workspace + workspaceSize;
 	unsigned tableLog;
 	unsigned maxSymbolValue = FSE_MAX_SYMBOL_VALUE;
+	size_t NCountLength;
+
+	workspace = counting + FSE_MAX_SYMBOL_VALUE + 1;
+	if ((BYTE *)workspace > wkspEnd)
```

Comment:
In theory, such comparison is UB, because to be true, `workspace` must be outside of valid memory object.
Would it be possible to convert this test
into a size comparison ?
Something like `if ( ((FSE_MAX_SYMBOL_VALUE+1)*sizeof(short)) > workspaceSize )` ?